### PR TITLE
Publish ironic-inspector endpoint via mDNS

### PR DESCRIPTION
**Pull request by dtantsur**

### CODE REVIEW

File: runironic-inspector.sh [hunk 1]
```diff
@@ -1,15 +1,27 @@
...
+if ! iptables -i $INTERFACE -p udp --dport 5353 -j ACCEPT > /dev/null 2>&1; then
+    iptables -i $INTERFACE -p udp --dport 5353 -j ACCEPT
+fi
+
 # Remove log files from last deployment
 rm -rf /shared/log/ironic-inspector
 
 mkdir -p /shared/log/ironic-inspector
 
+cp $CONFIG $CONFIG.orig
```
Comment (elfosardo) [inline runironic-inspector.sh:23]: nit: wrap second $CONFIG in braces

**Review by elfosardo — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -8,17 +8,22 @@ RUN mkdir -p /var/lib/ironic-inspector && \
...
     crudini --set /etc/ironic-inspector/inspector.conf processing ramdisk_logs_dir /shared/log/ironic-inspector/ramdisk && \
     crudini --set /etc/ironic-inspector/inspector.conf processing always_store_ramdisk_logs true && \
     crudini --set /etc/ironic-inspector/inspector.conf pxe_filter driver noop && \
     crudini --set /etc/ironic-inspector/inspector.conf processing node_not_found_hook enroll && \
     crudini --set /etc/ironic-inspector/inspector.conf discovery enroll_node_driver ipmi && \
-    crudini --set /etc/ironic-inspector/inspector.conf processing processing_hooks "\$default_processing_hooks,lldp_basic"
+    crudini --set /etc/ironic-inspector/inspector.conf processing processing_hooks "\$default_processing_hooks,lldp_basic" && \
+    # NOTE(dtantsur): keep this in sync with ironic-image/inspector.ipxe
+    # TODO(dtantsur): add ipa_inspection_collectors once we get ironic-lib 2.17.1
+    crudini --set /etc/ironic-inspector/inspector.conf mdns params 'ipa_debug:1,ipa_inspection_dhcp_all_interfaces:1,ipa_collect_lldp:1'
```
Comment (bfournie) [inline Dockerfile:26]: inspector.ipxe uses ipa-inspection-dhcp-all-interfaces and ipa-collect-lldp (i.e. dashes not underscores), does it matter?

Comment (dtantsur) [reply Dockerfile:26]: Yep :( These are different because of the way they're processed on the IPA side.

**Review by bfournie — APPROVED** [review]
Looks good.

**Pull request merged**